spec/knockout-fast-foreach-spec.js

```diff
@@ -327,34 +328,193 @@ describe("observable array changes", function () {
     assert.equal(target.text(), "123456")
   })
 
-  describe("isAdditionAdjacentToLast", function () {
-    it("is false for a deletion", function () {
-      assert.notOk(isAdditionAdjacentToLast(0, [{status: 'deleted'}]))
+  describe("DOM move capabilities", function() {
+    it("sorting complex data moves 1 DOM node", function() {
+      div = $("<div data-bind='fastForEach: obs'><div data-bind='html: testHtml'></div></div>");
+      ko.applyBindings(view, div[0]);
+      obs([{ id: 4, testHtml: '<span>A</span>' }, { id: 6, testHtml: '<span>B</span>' }, { id: 1, testHtml: '<span>C</span>' }])
+      var nodes = div.children().each(function(i) { this.id = i; }).toArray()
+      assert.equal(div.text(), 'ABC')
+      obs.sort(function(a, b) { return a.id - b.id; })
+      var nodes2 = div.children().toArray()
+      assert.equal(nodes[1].id, nodes2[2].id)
+      assert.equal(nodes[2].id, nodes2[0].id)
```

Comment:
Thanks, I wasn't aware of that, didn't use mocha before. In other tests I used custom node data for it, but I'll have a look and refactor it to be cleaner.